packages/plugins/plugin-assistant/src/stories/Query.stories.tsx

```diff
@@ -381,14 +382,55 @@ const ItemList = <T extends AnyEchoObject>({ items = [] }: { items?: T[] }) => {
   );
 };
 
+const FlushButton = ({ db }: { db?: EchoDatabase }) => {
```

Comment:
this would be much cleaner using a custom local hook pattern that provided the handler and state, with a regular button.

```tsx
const [state, icon, handleFlush] = useFlush();

return <IconButton onClick={handleFlush} />
```

Reply:
👍